### PR TITLE
Add different text for multiple projects

### PR DESCRIPTION
**Pull request by tomasMizera**
An option to make error messages more readable, not only: _invalid project_

image

### CODE REVIEW

File: app/projectsmodel.cpp [hunk 1]
```diff
@@ -48,14 +48,18 @@ void ProjectModel::findProjectFiles()
     projectFile.projectName = project.projectName;
     projectFile.projectNamespace = project.projectNamespace;
     QDateTime created = fi.created().toUTC();   // TODO: why UTC ???
-    if ( !project.qgisProjectFilePath.isEmpty() )
```
Comment (PeterPetrik) [inline app/projectsmodel.cpp:51]: I was thinking if LocalProjectInfo has new function `setInvalid(const std::string& reason)` or something similar, we can pass the error reason this way. this implementation does not set isValid == false, but the project is invalid. Also it uses projectFilePath as way how to pass the error around, which would not be required...

Comment (wonder-sk) [reply app/projectsmodel.cpp:51]: Good suggestion - I am easy, also happy to merge it as is... your call Peter - change it?

Comment (PeterPetrik) [reply app/projectsmodel.cpp:51]: if it is not too much struggle i would change it, my concern is that isValid() function with the current implementation does not say truth. and Tomas wants to hear the truth...

**Comment by tomasMizera** [comment]
@PeterPetrik @wonder-sk I changed the error propagation, what do you think about it now? `isValid` in `LocalProjectInfo` stays the same (as we discussed). This solution is up for discussion, attribute `projectError` needs to be reset on project sync - this bug is already on board.

File: app/localprojectsmanager.cpp [hunk 2]
```diff
@@ -28,6 +28,16 @@ static QString findQgisProjectFile( const QString &projectDir )
   {
     return foundProjectFiles.first();
   }
+  else if ( foundProjectFiles.count() > 1 )
+  {
+    // error: multiple project files found
+    err = QString( "ERR_MULTIPLE_PROJECTS" );
```
Comment (PeterPetrik) [inline app/localprojectsmanager.cpp:34]: just put the translated error message directly here?

File: app/localprojectsmanager.cpp [hunk 3]
```diff
@@ -28,6 +28,16 @@ static QString findQgisProjectFile( const QString &projectDir )
...
+  }
+  else
+  {
+    // no projects
+    err = QString( "ERR_NO_PROJECTS" );
```
Comment (PeterPetrik) [inline app/localprojectsmanager.cpp:39]: just put the translated error message directly here?

File: app/localprojectsmanager.h [hunk 4]
```diff
@@ -28,10 +28,17 @@ struct LocalProjectInfo
 {
   bool isValid() const { return !projectDir.isEmpty(); }
 
+  void setprojectFileError( const QString error ) { projectError = error; }
```
Comment (PeterPetrik) [inline app/localprojectsmanager.h:31]: remove function

File: app/localprojectsmanager.cpp [hunk 5]
```diff
@@ -109,8 +122,11 @@ void LocalProjectsManager::updateProjectStatus( const QString &projectDir )
 void LocalProjectsManager::addMerginProject( const QString &projectDir, const QString &projectNamespace, const QString &projectName )
 {
   LocalProjectInfo project;
+  QString err;
   project.projectDir = projectDir;
-  project.qgisProjectFilePath = findQgisProjectFile( projectDir );
+  project.qgisProjectFilePath = findQgisProjectFile( projectDir, err );
```
Comment (PeterPetrik) [inline app/localprojectsmanager.cpp:127]: send there directly `project. qgisProjectError`

File: app/localprojectsmanager.h [hunk 6]
```diff
@@ -28,10 +28,17 @@ struct LocalProjectInfo
...
+
+  bool isShowable() const { return projectError.isEmpty(); }
+
   QString projectDir;  //!< full path to the project directory
 
   QString qgisProjectFilePath;  //!< path to the .qgs/.qgz file (or empty if not have exactly one such file)
 
+  QString projectError; //!< If project is invalid, projectError carry more information why
```
Comment (PeterPetrik) [inline app/localprojectsmanager.h:39]: qgisProjectError

File: app/localprojectsmanager.cpp (hunk 2 above)
Comment (tomasMizera) [reply app/localprojectsmanager.cpp:34]: Sure, we can.

Comment (tomasMizera) [reply app/localprojectsmanager.cpp:34]: maybe not

**Pull request merged**